test/csi/e2e_test.sh

```diff
@@ -211,7 +218,7 @@ EOF
 
 # wait for pod predictor to be initialized
 repeat_cmd_until "kubectl get pod -n $KSERVE_TEST_NAMESPACE --selector='component=predictor' --output jsonpath='{.items[*].metadata.name}' | grep sklearn-iris-scenario-two | wc -l" "-gt 0" 60
-predictor_two=$(kubectl get pod -n $KSERVE_TEST_NAMESPACE --selector="component=predictor" --output jsonpath='{.items[1].metadata.name}')
+predictor_two=$(kubectl get pod -n $KSERVE_TEST_NAMESPACE --selector="component=predictor" --output jsonpath='{.items[0].metadata.name}')
```

Comment:
by working on this PR, I realized the test has a "bug" 
ie every Scenario after Scenario 1, it was assumed the Predictor pod for the current scenario was always the _second one_ ...not very good.

to avoid this problem, introduced after each Scenario on cleanup (when an Isvc KServe resource is being deleted) an explicit waiting for the terminating Pod to just be completely evicted.

This way, at the expense of a few seconds, we can ensure proper cleanup of resources at the end of each test scenario, and not rely on assumption/shortcuts in the following scenarios.